sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/SyncBenchmark.java

```diff
@@ -0,0 +1,359 @@
+                // TODO: support for other operationTypes will be added later
+//            case WriteLatency:
+//            case QueryInClauseParallel:
+//            case QueryCross:
+//            case QuerySingle:
+//            case QuerySingleMany:
+//            case QueryParallel:
+//            case QueryOrderby:
+//            case QueryAggregate:
+//            case QueryAggregateTopOrderby:
+//            case QueryTopOrderby:
+            case Mixed:
+                latency = metricsRegistry.timer("Latency");
+                break;
+            default:
+                break;
+        }
+
+        reporter.start(configuration.getPrintingInterval(), TimeUnit.SECONDS);
+        long startTime = System.currentTimeMillis();
+
+        AtomicLong count = new AtomicLong(0);
+        long i;
+
+        for ( i = 0; shouldContinue(startTime, i); i++) {
+
+            ResultHandler<T, Throwable> resultHandler = new ResultHandler<T, Throwable>() {
+                @Override
+                public T apply(T t, Throwable throwable) {
+                    successMeter.mark();
+                    concurrencyControlSemaphore.release();
+                    if (t != null) {
+                        assert(throwable == null);
+                        SyncBenchmark.this.onSuccess();
+                        synchronized (count) {
+                            count.incrementAndGet();
+                            count.notify();
+                        }
+                    } else {
+                        assert(throwable != null);
+
+                        failureMeter.mark();
+                        logger.error("Encountered failure {} on thread {}" ,
+                                     throwable.getMessage(), Thread.currentThread().getName(), throwable);
+                        concurrencyControlSemaphore.release();
+                        SyncBenchmark.this.onError(throwable);
+
+                        synchronized (count) {
+                            count.incrementAndGet();
+                            count.notify();
+                        }
+                    }
+
+                    return t;
+                }
+            };
+
+            concurrencyControlSemaphore.acquire();
+            final long cnt = i;
+
+            switch (configuration.getOperationType()) {
+                case ReadLatency:
+                    // TODO: support for other operation types will be added later
+//                case WriteLatency:
+//                case QueryInClauseParallel:
+//                case QueryCross:
+//                case QuerySingle:
+//                case QuerySingleMany:
+//                case QueryParallel:
+//                case QueryOrderby:
+//                case QueryAggregate:
+//                case QueryAggregateTopOrderby:
+//                case QueryTopOrderby:
+//                case Mixed:
+                    LatencyListener latencyListener = new LatencyListener(resultHandler, latency);
```

Comment:
Parameterized type, please change this to `LatencyListener<T>`... `new LatencyListener<>(resultHandler, ...)`

Reply:
addressed